### ISSUE TITLE
Improve performance so we can enable column sorting

### ISSUE THREAD

**Issue by moceap** [timlau/yumex-dnf#145]
Cant sort packages (by size, arch ... etc)

ver 4.3.2

**Comment by genodeftest** [comment]
The current situation is an intended workaround because our toolkit Gtk+ can't cope with a huge amount of packages to sort, at least not under python. If we made columns sortable, yumex-dnf would freeze for several seconds while sorting packages.

We should do some profiling on the sorting code and have a look if we can speed things up. I doubt that this will help though, unless we move to a non-interpreted programming language like C or Rust.

**Comment by moceap** [comment]
> The current situation is an intended workaround because our toolkit Gtk+ can't cope with a huge amount of packages to sort, at least not under python. If we made columns sortable, yumex-dnf would freeze for several seconds while sorting packages.

So can we make it as a choice in settings ?

**Comment by genodeftest** [comment]
> So can we make it as a choice in settings ?

Yes, we could, if someone does the work. In fact, yumex (for yum; the predecessor of yumex-dnf) had this feature.

**Comment by moceap** [comment]
If you meen yumex, it isn't!
